### PR TITLE
feat(IT Wallet): [SIW-1036] Add eID detail screen

### PR DESCRIPTION
**Pull request by gispada**
> [!WARNING]  
> This PR depends on #5780

## Short description

This PR adds the credential detail screen for the eID. The screen can support other credential types as well, it would just need a bit more fine tuning.

> [!Note]
> In the detail screen claims are grouped into sections, but this information is not included in the credential itself. To work around this, a mapping claim-section has been hardcoded in app. Claims that can not be assigned to any section are displayed after all the others.
> The **eID credential** used is also a **mock**.

## List of changes proposed in this pull request
- Extracted `RenderPidAssuranceLevel` and `RenderReleaserName` into separate components for easier reuse
- Added `ItwPresentationEidDetailScreen` component with related sub-components
- Modified `ItwCredentialClaim` to support hidden claims and dates with icon and badge
- Add utils in `itwClaimsUtils` to handle hardcoded claims configurations

## How to test
Navigate to **Profile > IT Wallet > Credential detail (eID)**. You should be able to see the eID detail screen.

## Preview

|iOS|Android|
|---|--------|
|<video src="https://github.com/pagopa/io-app/assets/52289599/03ab3b16-838f-4eb0-bbe4-006f1324692e">|<video src="https://github.com/pagopa/io-app/assets/52289599/7a7a7feb-f994-4669-9083-b398311d0086">|

### CODE REVIEW

**Review by dmnplb — CHANGES_REQUESTED** [review]
- `BaseScreenComponent` is a deprecated way to build the screen. I'll mark it as deprecated in a separate PR, but please use `useHeaderSecondLevel` instead. With this hook, you're setting the header through `react-navigation` instead of including it in a separate component (which is a bad practice)
- By not using `BaseScreenComponent`, you should avoid the white gap at the end of the screen on iPhone(s) without TouchID. Instead, use a proper `paddingBottom` on a `ScrollView`. See the attached screenshot for reference:



**EDIT:** I've just added a new hook to get the spacing value to apply as `paddingBottom`:
- https://github.com/pagopa/io-app/pull/5808

**Comment by gispada** [comment]
@dmnplb thanks for the `useScreenEndMargin` hook.

With regard to the `BaseScreenComponent`, I understand it is deprecated, but the screen was designed with a colored header and white icons. The `HeaderSecondLevel` doesn't support white icon buttons from what I've seen, they are fixed to color neutral. Am I right?

**Comment by dmnplb** [comment]
> With regard to the BaseScreenComponent, I understand it is deprecated, but the screen was designed with a colored header and white icons. The HeaderSecondLevel doesn't support white icon buttons from what I've seen, they are fixed to color neutral. Am I right?

@gispada That's correct. We need to add a way to render the `HeaderSecondLevel` with the `IconButton` color prop set to `contrast`. But the solution is not in the `BaseScreenComponent`. We should improve the `HeaderSecondLevel` with high priority

File: ts/features/itwallet/common/utils/__tests__/itwClaimsUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        {
+          label: "Numero di documento",
+          value: "XX1234567",
+          id: "document_number"
+        }
+      ],
+      noSection: [
+        { label: "Paese di rilascio", value: "IT", id: "issuing_country" },
+        {
+          label: "AutoritÃ  di rilascio",
```
Comment (hevelius) [inline ts/features/itwallet/common/utils/__tests__/itwClaimsUtils.test.ts:69]:
```suggestion
          label: "Autorità  di rilascio",
```

File: ts/features/itwallet/presentation/components/ItwClaimsSections.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+              title={I18n.t(
+                "features.itWallet.presentation.credentialDetails.personalDataTitle"
+              )}
+              canHideValues
+              claims={claims}
+            />
+          </>
+        ))
+      )}
+      {/* TODO: add section for mDL */}
```
Comment (hevelius) [inline ts/features/itwallet/presentation/components/ItwClaimsSections.tsx:41]: Is this TODO linked to a Jira story? What do you think to add a Jira ID if appropriate?

**Review by hevelius — CHANGES_REQUESTED** [review]

**Comment by hevelius** [comment]
> > With regard to the BaseScreenComponent, I understand it is deprecated, but the screen was designed with a colored header and white icons. The HeaderSecondLevel doesn't support white icon buttons from what I've seen, they are fixed to color neutral. Am I right?
> 
> @gispada That's correct. We need to add a way to render the `HeaderSecondLevel` with the `IconButton` color prop set to `contrast`. But the solution is not in the `BaseScreenComponent`. We should improve the `HeaderSecondLevel` with high priority

@dmnplb @gispada Not knowing the deadline to update the hook and instead having a close deadline for the wallet, I would suggest leaving this solution and adding a TODO task on jira for a future refactor.

File: ts/features/itwallet/common/utils/__tests__/itwClaimsUtils.test.ts (hunk 1 above)
Comment (gispada) [reply ts/features/itwallet/common/utils/__tests__/itwClaimsUtils.test.ts:69]: These tests are tied to the mocked credentials, and in the mDL that label is not properly encoded. If I change it with "attività" the test will fail.

Comment (hevelius) [reply ts/features/itwallet/common/utils/__tests__/itwClaimsUtils.test.ts:69]: You're right! It is a never resolved issue on the ISSUER side.

**Comment by dmnplb** [comment]
@hevelius @gispada Here's the enhancement applied to the `HeaderSecondLevel` that's needed to build the new screen without using the `BaseScreenComponent`:
- https://github.com/pagopa/io-app-design-system/pull/285

**Comment by gispada** [comment]
I updated the code to remove `BaseScreenComponent` and use `HeaderSecondLevel` with the new props from https://github.com/pagopa/io-app-design-system/pull/285. I also used the new hook `useScreenEndMargin` for the scroll view paddingBottom.

**Review by dmnplb — APPROVED** [review]

**Review by hevelius — COMMENTED** [review]
Should this extra space be here?

Simulator Screenshot - iPhone 15 Pro - 2024-06-14 at 12 10 54

**Comment by gispada** [comment]
> Should this extra space be here?

That space separates claims that are not mapped to a section from the other claims. I am not really sure how to handle this case, in the ideal world all claims should be mapped to a section. Moreover, in the mocked credential a few claims are missing or are extra compared to the design.

The mapping is configurable, we can adjust it as we start getting the real credentials or we can discuss another approach.

**Comment by hevelius** [comment]
> > Should this extra space be here?
> 
> That space separates claims that are not mapped to a section from the other claims. I am not really sure how to handle this case, in the ideal world all claims should be mapped to a section. Moreover, in the mocked credential a few claims are missing or are extra compared to the design.
> 
> The mapping is configurable, we can adjust it as we start getting the real credentials or we can discuss another approach.

the `VSpacer` is always added regardless of whether there was a claims section before or not. I believe a different approach should be used. For the moment, however, I would leave it like this.

**Review by hevelius — APPROVED** [review]

**Pull request merged**